plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1345,8 +1348,15 @@ public OptionalLong executeDelete(ConnectorSession session, ConnectorTableHandle
                 .deleteFromRowFilter(toIcebergExpression(handle.getEnforcedPredicate()))
                 .commit();
 
-        // TODO: it should be possible to return number of deleted records. https://github.com/trinodb/trino/issues/12055
-        return OptionalLong.empty();
+        Map<String, String> summary = icebergTable.currentSnapshot().summary();
+        String deletedRowsStr = summary.get(DELETED_RECORDS_PROP);
+        if (deletedRowsStr == null) {
+            return OptionalLong.empty();
```

Comment:
can we make Iceberg fill DELETED_RECORDS_PROP always?
let's file a PR and link with a TODO note here.